File: easy/3090_maximum_length_substring_with_two_occurrences/solution.cpp
#include <string>
#include <vector>

class Solution {
public:
  int maximumLengthSubstring(std::string s) {
    std::vector<int> charsCounter(26, 0);

    int maxLen = 0;

    for (size_t left{}, right{}; right < s.size(); ++right) {
      ++charsCounter[s[right] - 'a'];

      while (charsCounter[s[right] - 'a'] > 2)
        --charsCounter[s[left++] - 'a'];

      maxLen = std::max(maxLen, static_cast<int>(right - left + 1));
    }

    return maxLen;
  }
};

File: easy/3090_maximum_length_substring_with_two_occurrences/solution.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        charsCounter = [0] * 26

        maxLen = 1

        left = 0
        right = 0

        while right < len(s):
            idx = ord(s[right]) - 97
            charsCounter[idx] += 1

            while charsCounter[idx] > 2:
                charsCounter[ord(s[left]) - 97] -= 1
                left += 1

            maxLen = max(maxLen, right - left + 1)
            right += 1

        return maxLen
